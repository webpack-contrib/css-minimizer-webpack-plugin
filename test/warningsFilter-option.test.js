import postcss from "postcss";

import CssMinimizerPlugin from "../src/index";

import {
  compile,
  getCompiler,
  getErrors,
  getWarnings,
  readAssets,
} from "./helpers";

describe("warningsFilter option", () => {
  it('should match snapshot for a "function" value', async () => {
    const plugin = postcss.plugin("warning-plugin", () => (css, result) => {
      result.warn(`Warning from ${result.opts.from}`, {
        plugin: "warning-plugin",
      });
    });

    const compiler = getCompiler({
      entry: {
        foo: `${__dirname}/fixtures/test/foo.css`,
        bar1: `${__dirname}/fixtures/test/bar1.css`,
        bar2: `${__dirname}/fixtures/test/bar2.css`,
      },
    });

    new CssMinimizerPlugin({
      parallel: false,
      minify: (data) => {
        const [[fileName, input]] = Object.entries(data);

        return postcss([plugin])
          .process(input, { from: fileName, to: fileName })
          .then((result) => {
            return {
              code: result.css,
              map: result.map,
              error: result.error,
              warnings: result.warnings(),
            };
          });
      },
      warningsFilter(warning, file) {
        if (/foo/.test(file)) {
          return false;
        }

        return true;
      },
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });
});
