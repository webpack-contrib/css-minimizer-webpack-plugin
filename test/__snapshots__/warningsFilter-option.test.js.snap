// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`warningsFilter option should match snapshot for a "function" value: assets 1`] = `
Object {
  "bar1.css": "body {
  color: red;
}
body {
  color: red;
  font-size: 20px;
}
a {
  color: blue;
}

",
  "bar2.css": "h1 {
  color: green;
}
h2 {
  color: yellow;
}
h3 {
}

",
  "foo.css": "a {
  text-align: center;
}

",
}
`;

exports[`warningsFilter option should match snapshot for a "function" value: errors 1`] = `Array []`;

exports[`warningsFilter option should match snapshot for a "function" value: warnings 1`] = `
Array [
  "Warning: Css Minimizer Plugin: warning-plugin: Warning from bar1.css",
  "Warning: Css Minimizer Plugin: warning-plugin: Warning from bar2.css",
]
`;
